mindsdb/integrations/handlers/cohere_handler/cohere_handler.py

```diff
@@ -0,0 +1,134 @@
+import os
+from typing import Optional, Dict
+
+import cohere
+import pandas as pd
+
+from mindsdb.utilities.config import Config
+from mindsdb.integrations.libs.base import BaseMLEngine
+
+from mindsdb.utilities.log import get_log
+
+
+logger = get_log("integrations.cohere_handler")
+
+class CohereHandler(BaseMLEngine):
+    """
+    Integration with the Cohere Python Library
+    """
+    name = 'cohere'
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+
+        if 'using' not in args:
+            raise Exception("Cohere engine requires a USING clause! Refer to its documentation for more details.")
+
+        self.model_storage.json_set('args', args)
+
+    def predict(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+
+        args = self.model_storage.json_get('args')
+
+        input_keys = list(args.keys())
+
+        logger.info(f"Input keys: {input_keys}!")
+
+        input_column = args['using']['column']
+
+        if input_column not in df.columns:
+            raise RuntimeError(f'Column "{input_column}" not found in input data')
+        
+        result_df = pd.DataFrame() 
+
+        if args['using']['task'] == 'text-summarization':
+            result_df['predictions'] = df[input_column].apply(self.predict_text_summary)     
+
+        elif args['using']['task'] == 'text-generation':
+            result_df['predictions'] = df[input_column].apply(self.predict_text_generation)     
+
+        elif args['using']['task'] == 'language-detection':
+            result_df['predictions'] = df[input_column].apply(self.predict_language)     
+      
+        else:
+            raise Exception(f"Task {args['using']['task']} is not supported!")
+
+        result_df = result_df.rename(columns={'predictions': args['target']})
+        
+        return result_df
+
+
+    def _get_cohere_api_key(self, args, strict=True):
+        """ 
+        API_KEY preference order:
+            1. provided at model creation
+            2. provided at engine creation
+            3. COHERE_API_KEY env variable
+            4. cohere.api_key setting in config.json
+        """ 
+        # 1
+        if 'api_key' in args:
+            return args['api_key']
+        # 2
+        connection_args = self.engine_storage.get_connection_args()
+        if 'api_key' in connection_args:
+            return connection_args['api_key']
+        # 3
+        api_key = os.getenv('COHERE_API_KEY')
```

Comment:
If we provide option for setting up the key as env maybe we can add this to docs

Reply:
@ZoranPandovski this seems to be a recurring pattern (used in `openai` and potentially others). Perhaps let's open a refactoring issue so that the method can become a utility?